### PR TITLE
make TS definitions for DataSchemaValue more accurate

### PR DESCRIPTION
**Pull request by danielpeintner**

### CODE REVIEW

File: typescript/index.d.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ declare namespace WoT {
     export type ThingDescription = { [key: string]: any; };
 
 
-    export type DataSchemaValue = any;
+    export type DataSchemaValue = (string | number | any[] | object | null);
```
Comment (danielpeintner) [inline typescript/index.d.ts:80]: I am unsure about the array.

We could change `any[]` to `Array<Type>` where `Type` would be `any` or the list of types again? Not sure what makes more sense.

I am more in favor to use the `plain` array construct vs the class style...

Comment (relu91) [reply typescript/index.d.ts:80]: what's wrong with `DataSchemaValue[]`? ts cannot handle recursive declarations?

Comment (relu91) [reply typescript/index.d.ts:80]: https://www.typescriptlang.org/play?#code/C4TwDgpgBAIghsOBlAxgCwgWzgNTgGwFdoBeKACgGdgAnASwDsBzKAHygcMwCMIa3YCZOiy4CxANoBdAQHtuAKwgpgAzvnwBKANwAoXfgiqAbuIgAuQYlQZseIqSgSARBg2znU7UA

it seems to work

Comment (danielpeintner) [reply typescript/index.d.ts]: Good point. I missed the obvious :-(

**Comment by danielpeintner** [comment]
@zolkis new order update
= (null | boolean | number | string | object | DataSchemaValue[]);

**Comment by danielpeintner** [comment]
> @zolkis new order update
> = (null | boolean | number | string | object | DataSchemaValue[]);

Updated PR, please approve. Thanks!

**Review by relu91 — APPROVED** [review]

**Pull request merged**